vm/src/stdlib/signal.rs

```diff
@@ -0,0 +1,157 @@
+use crate::obj::objint::PyIntRef;
+use crate::pyobject::{IdProtocol, PyObjectRef, PyResult};
+use crate::vm::VirtualMachine;
+
+use std::sync::atomic::{AtomicBool, Ordering};
+
+use num_traits::cast::ToPrimitive;
+
+use nix::sys::signal;
+use nix::unistd::alarm as sig_alarm;
+
+// Signal triggers
+// TODO: 64
+const NSIG: usize = 15;
+
+static mut TRIGGERS: [AtomicBool; NSIG] = [
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+    AtomicBool::new(false),
+];
+
+extern "C" fn run_signal(signum: i32) {
+    unsafe {
+        TRIGGERS[signum as usize].store(true, Ordering::Relaxed);
+    }
+}
+
+fn signal(
+    signalnum: PyIntRef,
+    handler: PyObjectRef,
+    vm: &VirtualMachine,
+) -> PyResult<Option<PyObjectRef>> {
+    if !vm.isinstance(&handler, &vm.ctx.function_type())?
+        && !vm.isinstance(&handler, &vm.ctx.bound_method_type())?
+        && !vm.isinstance(&handler, &vm.ctx.builtin_function_or_method_type())?
+    {
+        return Err(vm.new_type_error("Hanlder must be callable".to_string()));
+    }
+    let signal = vm.import("signal", &vm.ctx.new_tuple(vec![]), 0)?;
+    let sig_dfl = vm.get_attribute(signal.clone(), "SIG_DFL")?;
+    let sig_ign = vm.get_attribute(signal, "SIG_DFL")?;
+    let signalnum = signalnum.as_bigint().to_i32().unwrap();
+    let signal_enum = signal::Signal::from_c_int(signalnum).unwrap();
+    let sig_handler = if handler.is(&sig_dfl) {
```

Comment:
@windelbouwman @coolreader18 any suggestion on how should I do this? This don't actually works...

Reply:
Ignore this. I had a typo...